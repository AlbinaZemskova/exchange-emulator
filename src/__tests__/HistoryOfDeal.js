import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import {
  HistoryOfDeals,
  mapStateToProps,
} from '../HistoryOfDeals';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<HistoryOfDeal />', () => {
  let wrapper;
  let history;

  beforeEach(() => {
    history = [
      {
        price: 1000,
        count: 200,
      },
      {
        price: 2000,
        count: 150,
      },
    ];
    wrapper = shallow(
      <HistoryOfDeals history={history} />,
    );
  });

  describe('HistoryOfDeal component', () => {
    describe('render', () => {
      it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      it('checking correct proptypes', () => {
        expect(() => (
          <HistoryOfDeals history={
            [
              {
                price: '1',
                count: '2',
              },
            ]
           }
          />
        )).toThrow();
      });
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the arrays of objects', () => {
      const mockState = {
        history: {
          history,
        },
      };

      const expected = {
        history,
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
