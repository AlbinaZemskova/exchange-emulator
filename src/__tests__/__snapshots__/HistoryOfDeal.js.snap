// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryOfDeal /> HistoryOfDeal component render renders without crashing 1`] = `
<Container
  tag="div"
>
  <Row
    className="text-center h4"
    tag="div"
  >
    <Col
      lg="12"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <h1
        className="border border-success p-2"
      >
        History Of Deal
      </h1>
      <div
        key="0"
      >
        200
          bitcoins sold for price   $
        1000
      </div>
      <div
        key="1"
      >
        150
          bitcoins sold for price   $
        2000
      </div>
    </Col>
  </Row>
</Container>
`;
